src/Compilers/Core/Portable/Symbols/SymbolComparer.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Allows for the comparison of two <see cref="ISymbol"/> instance based on certain criteria TODO
+    /// </summary>
+    public sealed class SymbolEqualityComparer : IEqualityComparer<ISymbol>
+    {
+        /// <summary>
+        /// Compares two <see cref="ISymbol"/> instances based on the default comparison rules, equivalent to calling <see cref="IEquatable{ISymbol}.Equals(ISymbol)"/>
+        /// </summary>
+        public static readonly SymbolEqualityComparer Default = new SymbolEqualityComparer(TypeCompareKind.AllNullableIgnoreOptions);
+
+        /// <summary>
+        ///Compares  two <see cref="ISymbol"/> instances, considering their nullability
```

Comment:
```suggestion
        /// Compares  two <see cref="ISymbol"/> instances, considering their nullability
```

Reply:
what does 'considering their nullability' mean?

Reply:
Filed #37384 to track better docs.